spec/features/reset_password_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe "Resetting passwords" do
+  let!(:user) { create :user, email: 'another@test.com', username: 'Another' }
+  let(:mail) { UserMailer.password_reset(user) }
+  before do
```

Comment:
I like to separate the setup and test stuff with an empty line between the `let`s, the `before do` and the `context`, but it's up to you.

Also, you shouldn't need the `let!(:user)` since it's being used in the before block it will get created regardless.